L78-python-rich-server-context.md

```diff
@@ -0,0 +1,73 @@
+Title
+----
+* Author(s): Chris St. Pierre
+* Approver: a11r
+* Status: Draft
+* Implemented in: python
+* Last updated: 2021-02-17
+* Discussion at: <google group thread> (filled after thread exists)
+
+## Abstract
+
+Expose context status code, details, and trailing metadata on the
+server side.
+
+## Background
+
+Currently, on the server-side `ServicerContext` object, status code,
+details, and trailing metadata are private. As a result, an
+interceptor cannot take action based on any of those properties.
+
+### Related Proposals
+
+N/A
+
+## Proposal
+
+We will add three methods to the servicer context objects:
+
+* `code()` returns the status code.
+* `details()` returns the state details.
+* `trailing_metadata()` returns the trailing metadata.
+
+All three are functions (not properties) for symmetry with the stub
+context class.
+
+## Rationale
+
```

Comment:
Let me trace out an internal dialog I had when reading your proposal. It would probably be good to add this line of reasoning to the gRFC as some color.

Why add getter methods to the `ServicerContext` when the author of the service handler is the one that calls the corresponding setter methods? They could keep track of what they set themself.  Interceptor authors, however, don't have this option. They're not necessarily in control of the service handler. So it makes sense to make these getters available to them.

But once we add the getters to the interceptors, we want them added to the non-interceptor `ServicerContext` to maintain uniformity between the two interfaces.